packages/frontend/src/pages/UserDetails/UserDetailsPage.tsx

```diff
@@ -0,0 +1,88 @@
+import { faUserGroup } from '@fortawesome/free-solid-svg-icons';
+import BreadCrumb from '@/components/BreadCrumb';
+import BackLink from '@/navigation/BackLink';
+import { SingleUser, SingleUserParams, useAdminUsers } from '@/model/users';
+import { useRecoilValue } from 'recoil';
+import { useParams } from 'react-router-dom';
+import { formatIsoDateUTC } from '@/utils/date';
+import { shortenEthAddress } from '@/utils/index';
+import { UserDto, UserRole } from 'api/dist/user/types';
+import { toast } from 'react-hot-toast';
+
+const roles = [UserRole.ADMIN, UserRole.FORWARDER, UserRole.QUANTIFIER];
+
+const UserDetailsPage = (): JSX.Element => {
+  const { userId } = useParams<SingleUserParams>();
+  const user = useRecoilValue(SingleUser({ userId }));
+  const { addRole, removeRole } = useAdminUsers();
+
+  const handleRole = (role: UserRole, user: UserDto): void => {
+    let resp;
+    const isRemove = user.roles.includes(role);
+    async (): Promise<void> => {
+      if (isRemove) {
+        resp = await removeRole(user._id, role);
+      } else {
+        resp = await addRole(user._id, role);
+      }
+      if (resp?.status === 200) {
+        toast.success(`Role ${isRemove ? 'removed' : 'added'} successfully!`);
+      }
+    };
+  };
+
+  return (
+    <div className="max-w-2xl mx-auto">
+      <BreadCrumb name="User details" icon={faUserGroup} />
+      <BackLink to="/pool" />
+      <div className="praise-box flex flex-col gap-2">
+        <span>User identity</span>
+        {!!user && (
+          <>
+            <span className="text-xl font-bold">
+              {shortenEthAddress(user.ethereumAddress || '')}
+            </span>
+            <span>Created: {formatIsoDateUTC(user.createdAt)}</span>
+            <span>Last updated: {formatIsoDateUTC(user.updatedAt)}</span>
+          </>
+        )}
+      </div>
+      <div className="praise-box flex flex-col gap-2">
+        <span>Linked Discord identities</span>
+        {user?.accounts?.map((account, index) => (
+          <>
+            <span>User account #{index + 1}</span>
+            <span className="text-xl font-bold">{account.name}</span>
+            <span>Discord User ID: {account.user}</span>
+            <span>Created: {formatIsoDateUTC(account.createdAt)}</span>
+            <span>Last updated: {formatIsoDateUTC(account.updatedAt)}</span>
```

Comment:
Can we tighten the spacing of the last 3 lines slightly? To better reflect the mockup.

Reply:
Mockup in Miro doesn't allow for any custom line spacing. Avoid `<span>` when not explicitly needed. In this case you can get default line height by wrapping in a `<div>` instead and add linebreaks `<br>`.